cpp_modules/src/mcts/Mcts.cpp

```diff
@@ -43,6 +43,33 @@ auto Mcts::randomChoice(const Container &container) -> decltype(*std::begin(cont
 	return *it;
 }
 
+Action Mcts::weightedChoice(const std::vector<Action> &actions) {
+	std::vector<double> weights(actions.size());
```

Comment:
Yes they are. 

for 1 base there is most often 4 mineral fields and 2  vespene geysers.
meaning their will be 16 workers getting minerals, and 6 workers collecting vespene adding up to 22 workers per base.

Since the base usually has 2 vespene geysers we should build 2 collectors per base.

Each base adds 15 to the population limit, and a house adds 8. 1 house and 1 base provides enough to support the 22 workers that should be at each base.